### DISCUSSION TITLE
Mocking onetable using jest in nodejs

### DISCUSSION THREAD

**Discussion by mishra-amrita** [sensedeep/dynamodb-onetable#96] [Q&A] [answered]
Can someone shed some light on how to mock the Table class in jest and its underlying methods for the get/create/fetch etc.
I have some code in a handler that gets a user record out of my dynamodb table. And we are using the getModel method to pull out the specific model and then perform a get operation on the model.

**Comment by mobsense** [comment, marked as answer]
Have a look at our Jest unit tests.

We run a local Dynamo (dynamo-db-local) and simply start the process and end for each unit test. Seems to run pretty quickly and gives you a very clean environment for each test.  

This way you get the real OneTable for your tests without the overhead of creating mocks.
